packages/discovery-react-components/src/components/SearchFacets/SearchFacets.tsx

```diff
@@ -131,22 +130,21 @@ const SearchFacets: FC<SearchFacetsProps> = ({
     (componentSettings && componentSettings.aggregations) ||
     [];
 
-  const searchParamsAggregationChanged = useCompare(searchParameters.aggregation);
   useEffect(() => {
     async function fetchData() {
-      setFetchState('loading');
+      setFetchAggregationState(FetchAggregationStates.LOADING);
```

Comment:
I think it makes sense to keep it in the `SearchFacets` component since this is specific to the state for the facets/aggregations.

Reply:
instead of setting the fetch aggregation state here, can we just convert this to using the `useDataApi` similar to how we do it for search results?  https://github.com/watson-developer-cloud/discovery-components/blob/master/packages/discovery-react-components/src/utils/useDataApi.ts#L239-L277

something like
```typescript
export const useGlobalAggregationsApi = (
  searchParameters: DiscoveryV2.QueryParams,
  searchClient: SearchClient
): [GlobalAggregationsResponseStore, GlobalAggregationsStoreActions] => {
  const {
    state: aggregationState,
    parameters: currentGlobalAggregationParameters,
    setParameters: setGlobalAggregationParameters,
    setData: setGlobalAggregationsResponse,
    setFetchToken
  } = useDataApi(searchParameters, null, searchClient.query, searchClient);

  return [
    {
      ...aggregationState,
      parameters: currentGlobalAggregationParameters
    },
    {
      setGlobalAggregationParameters,
      setGlobalAggregationsResponse,
    }
  ];
};
```

then use it like
```typescript
const [globalAggregationsResponseStore, { fetchGlobalAggregations, setGlobalAggregationsParameters }] = useGlobalAggregationsApi(
    { ...fetchAggregationsResponseStoreDefaults.parameters, projectId },
    searchClient
  );
```

make sure to add the `fetchGlobalAggregations` as one of the api methods in the `SearchContext` as well as `globalAggregationsResponseStore` as part of the `SearchContextApi`

then you can just check the fetch state in the SearchFacets component like
```
const { fetchGlobalAggregations } = useContext(SearchApi);
const { globalAggregationsResponseStore: { isLoading, isError, data }} = useContext(SearchContext);
```

Reply:
Yes, good point and thank you for the details. I've updated to use the `useDataApi` too 👍